src/wbase/Task.cc

```diff
@@ -427,13 +427,19 @@ wpublish::QueryStatistics::Ptr Task::getQueryStats() const {
 }
 
 /// Flag the Task as cancelled, try to stop the SQL query, and try to remove it from the schedule.
-void Task::cancel() {
+void Task::cancel(bool logIt) {
     if (_cancelled.exchange(true)) {
         // Was already cancelled.
         return;
     }
 
-    LOGS(_log, LOG_LVL_DEBUG, "Task::cancel " << getIdStr());
+    if (logIt) {
+        if (!_ujData->getCancelled()) {
+            LOGS(_log, LOG_LVL_DEBUG, "Task::cancel " << getIdStr() << " UberJob still live.");
+        } else {
+            LOGS(_log, LOG_LVL_TRACE, "Task::cancel " << getIdStr());
```

Comment:
Just curious. What's the reason behind using different logging levels for checking the cancellation status of an object?

Reply:
Mostly, it's trying to keep the log files under control. There can be tens of thousands of `Tasks` on a worker for a single User Query. If the user cancelled the query, or the Tasks are being cancelled because LIMIT has been reached, is really unlikely that the cancel log message will be useful and there will be a huge number of them.

On the other hand, `cancel()` being called for any other reason could be incredibly useful for debugging. Trying to keep the useful log messages without the junk messages is a bit of a trick.